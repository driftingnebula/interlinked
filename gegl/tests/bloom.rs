use {
  gegl::{Bloom, GeglOperation},
  insta::assert_debug_snapshot,
};

#[test]
fn test_bloom_graph() {
  let mut graphs = vec![];

  let op = Bloom {
    limit_exposure: true,
    radius: 100_f64,
    softness: 100_f64,
    ..Default::default()
  };

  let default = Bloom::default();

  graphs.push(("non-default: exclude defaults", op.graph(false)));
  graphs.push(("non-default: include defaults", op.graph(true)));
  graphs.push(("default: exclude defaults", default.graph(false)));
  graphs.push(("default: include defaults", default.graph(true)));

  assert_debug_snapshot!("graphs", graphs);
}
