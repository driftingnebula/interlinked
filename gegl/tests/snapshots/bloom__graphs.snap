---
source: gegl/tests/bloom.rs
expression: graphs
---
[
    (
        "non-default: exclude defaults",
        [
            "gegl:bloom",
            "limit-exposure=true",
            "radius=100",
            "softness=100",
        ],
    ),
    (
        "non-default: include defaults",
        [
            "gegl:bloom",
            "limit-exposure=true",
            "radius=100",
            "softness=100",
            "strength=50",
            "threshold=50",
        ],
    ),
    (
        "default: exclude defaults",
        [
            "gegl:bloom",
        ],
    ),
    (
        "default: include defaults",
        [
            "gegl:bloom",
            "limit-exposure=false",
            "radius=10",
            "softness=25",
            "strength=50",
            "threshold=50",
        ],
    ),
]
